File: src/petlja/os7-202324-kvalifikacije1n/F.py
from math import isqrt

def sqr(x):
    return x*x

p = int(input())

x = 1
a = 2
aa = 4

while sqr(isqrt(p)) < p and aa <= p:
    while p % aa == 0:
        x *= a
        p //= aa
    if p % a == 0:
        p //= a
    a += 1
    aa = a*a

if sqr(isqrt(p)) == p:
    x *= isqrt(p)

print(x)

File: src/codeforces/2244/F.pas
program _F;
{$MODE DELPHI}{$OPTIMIZATION LEVEL3,ON}
// #unsolved
uses
	math;
var
	notc, tci, n, i, a, mn, mx: int32;
	InputBuf, OutputBuf: array [1 .. 65536] of Char;

begin
	SetTextBuf(Input, InputBuf);
	SetTextBuf(Output, OutputBuf);

	readln(notc);
	for tci := 1 to notc do begin

		readln(n);

		for i := 1 to n do begin
			read(a);
		end;
		readln;

		writeln((mx - mn + 1) div 2);

	end;
end.
